contrib/pkg/deprovision/awstagdeprovision.go

```diff
@@ -25,6 +31,54 @@ func NewDeprovisionAWSWithTagsCommand() *cobra.Command {
 				log.WithError(err).Error("Cannot complete command")
 				return
 			}
+
+			// If environment variables are not set, assume we're running in a deprovision pod with an aws credentials secret mounted.
+			// Parse credentials from files mounted in as a secret volume and set as env vars. We use the file
+			// instead of passing env vars directly so we can monitor for changes and restart the pod if an admin
+			// modifies the creds secret after the uninstall job has launched.
+			if os.Getenv("AWS_ACCESS_KEY_ID") == "" && os.Getenv("AWS_SECRET_ACCESS_KEY") == "" {
```

Comment:
If we are changing the way that we are getting creds, I would **love** to support a creds file to keep parity with the `hiveutil create-cluster` command.

Reply:
Our established AWS secret format does not fit the known AWS cred file formats. So to do this we'd have to start generating another secret containing the file contents. That was complexity I'm not super eager to get into.

Reply:
I am not suggesting that we change the AWS secret format (although I **do** think that is a good long-term change). I am suggesting that running `hiveutil aws-tag-deprovision` directly should support using an AWS creds file.

Reply:
We made a similar change in the last few months for the GCP deprovision because of users' complaints.

Reply:
I think as part of starting up the uninstall, the code can take the contents of the secret and make a minimal AWS creds file that we can point to with `AWS_SHARED_CREDENTIALS_FILE` pointing to a file that looks like

```
[default]
aws_access_key_id = ACCESS_ID_FROM_SECRET
aws_secret_access_key = SECRET_KEY_FROM_SECRET
```